sos/plugins/navicli.py

```diff
@@ -67,21 +71,35 @@ def get_navicli_SP_info(self, SP_address):
 
     def setup(self):
         self.get_navicli_config()
+        ipaddr_list = self.get_option_as_list("ipaddrs", delimiter = " ")
         CLARiiON_IP_address_list = []
-        CLARiiON_IP_loop = "stay_in"
-        while CLARiiON_IP_loop == "stay_in":
-            try:
-                ans = input("CLARiiON SP IP Address or [Enter] to exit: ")
-            except:
-                return
-            if self.check_ext_prog("navicli -h %s getsptime" % (ans,)):
-                CLARiiON_IP_address_list.append(ans)
+
+        if len(ipaddr_list) == 0:
+            if not self.batch:
+                CLARiiON_IP_address_list = []
+                CLARiiON_IP_loop = "stay_in"
+                while CLARiiON_IP_loop == "stay_in":
+                    try:
+                        ans = input("CLARiiON SP IP Address or [Enter] to exit: ")
+                    except:
+                        return
+                    if self.check_ext_prog("navicli -h %s getsptime" % (ans,)):
+                        CLARiiON_IP_address_list.append(ans)
+                    else:
+                        if ans != "":
+                            print("The IP address you entered, %s, is not to an "
+                                  "active CLARiiON SP." % ans)
+                        if ans == "":
+                            CLARiiON_IP_loop = "get_out"
             else:
-                if ans != "":
-                    print("The IP address you entered, %s, is not to an "
-                          "active CLARiiON SP." % ans)
-                if ans == "":
-                    CLARiiON_IP_loop = "get_out"
+                self._log_warn("No CLARiiON SP IP Address specified as plugin option and in batch mode: skipping")
+        else:
+            for SP_address in ipaddr_list:
+                if self.check_ext_prog("navicli -h %s getsptime" % (SP_address,)):
+                    CLARiiON_IP_address_list.append(SP_address)
+                else:
+                    self._log_warn("IP address %s is not an active CLARiiON SP." % SP_address)
+
         # Sort and dedup the list of CLARiiON IP Addresses
         CLARiiON_IP_address_list.sort()
         for SP_address in CLARiiON_IP_address_list:
```

Comment:
I'm not sure this is the approach I would want to take for this. When this was discussed previously, the design I was aiming for was to add a new option API, that allows options to either be supplied from the command line (i.e. the same as `get_option()` currently does, or, if the value is unfilled, and we are not in batch mode, to then prompt the user for the value (with a caller-supplied prompt).

I can see this mechanism being useful in numerous plugins.